functions/go/set-image/transformer/images_transformer.go

```diff
@@ -0,0 +1,170 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/custom"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/image_util"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+// SetImage supports the set-image workflow, it uses Config to parse functionConfig, Transform to change the image
+type SetImage struct {
+	// Image is the desired image
+	Image image_util.Image `json:"image,omitempty" yaml:"image,omitempty"`
+	// ConfigMap keeps the data field that holds image information
+	ConfigMap map[string]string `json:"data,omitempty" yaml:"data,omitempty"`
+	// ONLY for kustomize, AdditionalImageFields is the user supplied fieldspec
+	AdditionalImageFields image_util.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+	// context logs each detailed result
+	context *fn.Context
+	// resultCount logs the total count image change
+	resultCount int
+}
+
+// Config transforms the data from ConfigMap to SetImage struct
+func (imagetrans *SetImage) Config() error {
+	for key, val := range imagetrans.ConfigMap {
+		switch key {
+		case "name":
+			imagetrans.Image.Name = val
+		case "newName":
+			imagetrans.Image.NewName = val
+		case "newTag":
+			imagetrans.Image.NewTag = val
+		case "digest":
+			imagetrans.Image.Digest = val
+		default:
+			return fmt.Errorf("SubObject has unmatched field type: `data` with wrong field name %v", key)
```

Comment:
I think we already talked about this before (can't find my previous comment). but it does not really make sense to the users about what `SubObject` (or `KubeObject`) means and how they can fix it.   Is there any specific reason to use `SubObject` here?

Reply:
I see. I used this phrase to keep the same format with previous unit test result. I agree I should give a better message.